prisma/schema.prisma

```diff
@@ -1,39 +1,46 @@
 generator client {
-  provider = "prisma-client-js"
+  provider      = "prisma-client-js"
   binaryTargets = ["native", "linux-musl"]
 }
 
 datasource db {
-  provider = "mysql"
-  url      = env("DATABASE_URL")
+  provider          = "mysql"
+  url               = env("DATABASE_URL")
   shadowDatabaseUrl = env("SHADOW_DATABASE_URL")
 }
 
 model PaybuttonAddress {
-  id          Int        @id @default(autoincrement())
-  address     String     @db.VarChar(255)
-  createdAt   DateTime   @default(now())
-  updatedAt   DateTime   @updatedAt
+  id          Int       @id @default(autoincrement())
+  address     String    @db.VarChar(255)
+  createdAt   DateTime  @default(now())
+  updatedAt   DateTime  @updatedAt
   chainId     Int
   paybuttonId Int
   chain       Chain     @relation(fields: [chainId], references: [id], onUpdate: Restrict)
-  paybutton  Paybutton @relation(fields: [paybuttonId], references: [id], onDelete: Cascade, onUpdate: Restrict)
+  paybutton   Paybutton @relation(fields: [paybuttonId], references: [id], onDelete: Cascade, onUpdate: Restrict)
+
+  @@index([chainId], map: "PaybuttonAddress_chainId_fkey")
+  @@index([paybuttonId], map: "PaybuttonAddress_paybuttonId_fkey")
 }
 
 model Paybutton {
-  id                  Int                   @id @default(autoincrement())
-  providerUserId      String?               @db.VarChar(255)
-  createdAt           DateTime              @default(now())
-  updatedAt           DateTime              @updatedAt
-  addresses           PaybuttonAddress[]
+  id             Int                  @id @default(autoincrement())
+  name           String               @db.VarChar(255)
+  uuid           String               @default(dbgenerated("uuid()"))
+  buttonData     String               @db.LongText
```

Comment:
What were you thinking we put in here?

My original idea for `buttonData` was tied to specific payments but I'm thinking `paymentData` makes more sense for that anyway. `paymentData` would also be used as the data container for BIP 70 transactions.

Reply:
I didn't think about anything specific. If it should be only for payments, then `paymentData` seems like a good idea, but it's less generic, so if we want to allow users to store another kind of data in the future, storing it as a key of `paymentData` would be weird, while it would be easy to store something like `buttonData = {'paymentData': ..., 'otherData': ...}`

Reply:
The thing is that it still makes sense IMO to put any sort of payment or button info in there if it's done at the payment level (like we do on Coin Dance). But what this is in the code is at the button level -- something that's set once -- which is different.

Reply:
I see. I implemented this thinking about something that is done at the button level (not necessarily only once, since we will probably have an "Edit" page). If it is done in the payment level, I think we would best save it in another table, something related to payments (e.g. if we start tracking payments as rows in some "PaybuttonPayment" table)

Reply:
Yeah ok. I'm good with having both.